elixir/apps/domain/test/domain/auth/adapters/userpass_test.exs

```diff
@@ -50,7 +50,7 @@ defmodule Domain.Auth.Adapters.UserPassTest do
 
       assert errors_on(changeset) == %{
                provider_virtual_state: %{
-                 password: ["should be at least 12 character(s)"],
+                 password: ["should be at least 12 byte(s)"],
```

Comment:
So curious regarding this change 😆

Reply:
@jamilbk Some hashing functions (like bcrypt) are having a hard limit in bytes, not characters (UTF can span multiple bytes). We use Argon which has a limit of 4096 bytes, but I decided to keep a low realistic limit in there and make sure it's calculated properly for the future so that we don't get confused by it later.

Reply:
And I did not bother about UI complications because I believe we don't plan to use userpass in production.